File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  switch(true) {
    case (input.toUpperCase() === input) && (input.toLowerCase() !== input):
      return 'Whoa, chill out!'; 
    case (input.charAt((input.length-1)) === '?'):
      return 'Sure.';
      case (/\S/.test(input) === false):
      return 'Fine. Be that way!';
    case (true):
      return 'Whatever.';
  };
};

module.exports = Bob;

File: java/bob/src/main/java/Bob.java
public class Bob {

    public String hey(String input) {
        if (input.equals("")) {
            return "Fine. Be that way!";
        }
        if (input.equals(input.toUpperCase()) && !input.equals(input.toLowerCase())) {
            return "Whoa, chill out!";
        }
        if (input.endsWith("?")) {
            return "Sure.";
        }
        return "Whatever.";
    }

}
